### PR TITLE
fix headers() and cookies()

### PR DESCRIPTION
**Pull request by vetinari**
they were not serialized properly

### CODE REVIEW

**Comment by mohabusama** [comment]
Please add the exception log and error tag to ``post_trial_run`` failure. Also would be useful to add span tag ``trial_run=True`` for ``worker_task_processing`` in case of trial run.

**Comment by vetinari** [comment]
the tracing should be a different PR, right :)

File: zmon_worker_monitor/zmon_worker/tasks/main.py [hunk 1]
```diff
@@ -1747,6 +1751,8 @@ def notify_for_trial_run(self, val, req, alerts, force_alert=False):
             try:
                 is_in_period = in_period(alert.get('period', ''))
             except InvalidFormat, e:
+                current_span.log_kv('exception', traceback.format_exc())
```
Comment (mohabusama) [inline zmon_worker_monitor/zmon_worker/tasks/main.py]: ``log_kv({'exception': trace...})``

File: zmon_worker_monitor/zmon_worker/tasks/main.py [hunk 2]
```diff
@@ -1747,6 +1751,8 @@ def notify_for_trial_run(self, val, req, alerts, force_alert=False):
...
+                current_span.set_tag('error', True)
```
Comment (mohabusama) [inline zmon_worker_monitor/zmon_worker/tasks/main.py]: Let's not set this as an error, I would rather add another tag, like, ``invalid_timeperiod=True`` so we can isolate those while searching.

File: zmon_worker_monitor/workflow.py [hunk 3]
```diff
@@ -215,6 +215,9 @@ def process_message(queue, known_tasks, reactor, msg_obj, current_span):
     timelimit = msg_body.get('timelimit')  # [90, 60]
     t_hard, t_soft = timelimit
 
+    if taskname == 'trial_run':
```
Comment (mohabusama) [inline zmon_worker_monitor/workflow.py]: May be we can set a tag ``set_tag('task_name', taskname)``, that way we can search and isolate all operations/tasks.

File: zmon_worker_monitor/zmon_worker/tasks/main.py [hunk 4]
```diff
@@ -1779,6 +1787,8 @@ def notify_for_trial_run(self, val, req, alerts, force_alert=False):
 
         # TODO: except SoftTimeLimitExceeded:
         except Exception:
+            current_span.log_kv('exception', traceback.format_exc())
```
Comment (mohabusama) [inline zmon_worker_monitor/zmon_worker/tasks/main.py]: ``log_kv({'exception': ...})``

File: zmon_worker_monitor/zmon_worker/tasks/main.py [hunk 5]
```diff
@@ -1761,6 +1767,8 @@ def notify_for_trial_run(self, val, req, alerts, force_alert=False):
                 }
                 result_json = json.dumps(result, cls=JsonDataEncoder)
             except TypeError, e:
+                current_span.log_kv('exception', traceback.format_exc())
```
Comment (mohabusama) [inline zmon_worker_monitor/zmon_worker/tasks/main.py]: ``log_kv({'exception': ...})``

File: zmon_worker_monitor/zmon_worker/tasks/main.py [hunk 6]
```diff
@@ -1731,12 +1732,15 @@ def post_trial_run(self, id, entity, result):
                              data=json.dumps(val, cls=JsonDataEncoder),
                              headers=headers)
             except Exception:
+                current_span.log_kv('exception', traceback.format_exc())
```
Comment (mohabusama) [inline zmon_worker_monitor/zmon_worker/tasks/main.py]: log_kv({'exception': ...})

File: zmon_worker_monitor/zmon_worker/tasks/main.py [hunk 7]
```diff
@@ -1779,6 +1786,8 @@ def notify_for_trial_run(self, val, req, alerts, force_alert=False):
 
         # TODO: except SoftTimeLimitExceeded:
         except Exception:
+            current_span.log_kv({'exception', traceback.format_exc()})
```
Comment (mohabusama) [inline zmon_worker_monitor/zmon_worker/tasks/main.py]: ``{'exception': ...}`` iso ``{'exception', ...}``
Dict not Set 😉

File: Dockerfile [hunk 8]
```diff
@@ -3,13 +3,11 @@ FROM registry.opensource.zalan.do/stups/ubuntu:latest
 #making this a cachable point as compile takes forever without -j
 
 RUN apt-get update && apt-get -y install python-pip python-dev libev4 libev-dev python-psycopg2 libpq-dev libldap2-dev libsasl2-dev libssl-dev libsnappy-dev iputils-ping freetds-dev git && \
-    pip2 install -U pip setuptools urllib3 Cython
+    pip2 install -U pip setuptools urllib3 Cython opentracing-utils
```
Comment (mohabusama) [inline Dockerfile]: May be we can move ``opentracing-utils`` to ``requirements.txt`` with the rest of "normal" packages.

File: zmon_worker_monitor/zmon_worker/tasks/main.py (hunk 7 above)
Comment (mohabusama) [reply zmon_worker_monitor/zmon_worker/tasks/main.py]: Same for the rest

**Comment by vetinari** [comment]
:+1:

**Comment by mohabusama** [comment]
👍

**Pull request merged**